spec/features/course/homepage_spec.rb

```diff
@@ -6,10 +6,23 @@
 
   with_tenant(:instance) do
     let(:course) { create(:course, :opened) }
-    let(:achievement_feed_notification) do
-      achievement = create(:course_achievement)
+    let(:feed_notifications) do
```

Comment:
Totally agree, I think the difficulty is to come up with a set of guidelines to say "when these sets of conditions are met, we split it; otherwise it should be merged into one."